### PR TITLE
Refactor: Parallelize Investigation via LangGraph Send()

### PR DESCRIPTION
**Pull request by Jeel3011**
**Type of Change:**
- [x] Feature
- [ ] Bug Fix
- [x] Improvement (Refactor / Performance)

**Description:**
This PR resolves a significant performance bottleneck in the `investigate` loop by migrating from sequential action execution to a parallelized, native LangGraph Fan-out/Fan-in architecture using `langgraph.constants.Send`.

*Changes:*
- Split the monolithic `node_investigate` into `node_investigate_hypothesis` (executing a single action) and `merge_hypothesis_results`.
- Implemented the `distribute_hypotheses` router to dispatch parallel graph branches based on the `planned_actions` array.
- Updated `AgentState` with a custom `merge_results_reducer` to properly combine isolated subgraph `hypothesis_results` concurrently.
- Re-wired `graph.py` to route through these new parallel nodes natively.

**Testing:**
- [x] Added `tests/nodes/test_parallel_investigate.py` to verify proper sub-graph routing distribution.
- [x] Ran all local checks (`make lint`, `make typecheck`, `make test-cov`); the entire test suite passes.
- [x] Verified graph compilations and state synchronizations remain intact.

**Impact Analysis:**
- *Backward compatible?* Yes. The `AgentState` schema changes are backwards-compatible additions (`action_to_run` and `hypothesis_results`), and the final state mutation behaves identically to the old sequential node.
- *Performance impact:* Highly positive. Actions that depend on external APIs (e.g., Datadog, Prometheus) now execute concurrently in the LangGraph lifecycle.

**AI-Assisted PRs:**
- [x] I reviewed every single line of AI-generated code.
- [x] I understand the logic and can explain it in my own words.
- [x] I tested edge cases (e.g., handling empty `planned_actions`).
- [x] I modified output to match project conventions (Ruff formatting, MyPy types).
- [x] Verified tests pass with the code.

### CODE REVIEW

**Comment by Jeel3011** [comment]
fixed it , check it

File: tests/nodes/test_parallel_investigate.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+"""Tests for parallel investigate hypothesis routing and merging."""
+
+import pytest
+from langgraph.constants import Send
+
+from app.nodes.investigate.execution.execute_actions import ActionExecutionResult
+from app.nodes.investigate.merge import merge_hypothesis_results
```
[the remark this answers is not in the document]
Comment (VaibhavUpreti) [reply tests/nodes/test_parallel_investigate.py:7]: @Jeel3011 could you please address these?

**Review by VaibhavUpreti — APPROVED** [review]
awesome @Jeel3011 , thanks a lot for adding this, welcome to the OpenSRE community
